### PR TITLE
Rename entity_spec and make it work in non-entity modules

### LINKED ISSUE #404: make the entity_spec work for non-entity modules
##### SUMMARY

See https://github.com/theforeman/foreman-ansible-modules/pull/335#discussion_r312357992

TL;DR: entity_spec is currently coupled to the fact that it describes an "entity", while we have action modules that don't operate on an entity as such, but would still benefit from the conversion done in the spec. maybe we should call it foreman_spec?

##### ISSUE TYPE
 - Feature Idea

### PR DESCRIPTION
**Pull request by nesanton**
* Rename entity_spec to foreman_spec
* Make foreman_spec usable by ForemanAnsibleModule class

I'm sure this is too rookie of an attempt to fix #404, but I wanted to begin

### CODE REVIEW

File: tests/test_entity_spec_helper.py [hunk 1]
```diff
@@ -1,11 +1,11 @@
-from plugins.module_utils.foreman_helper import _entity_spec_helper
```
Comment (mdellweg) [inline tests/test_entity_spec_helper.py]: That file could be renamed, too.

Comment (nesanton) [reply tests/test_entity_spec_helper.py]: oh right

**Comment by nesanton** [comment]
Idk if `entity_dict` still makes sense in `entity_dict = module.clean_params()`

**Comment by nesanton** [comment]
> Idk if `entity_dict` still makes sense in `entity_dict = module.clean_params()`

when I look at it it makes less and less sense. Should we call it something like `module_params` everywhere?

**Comment by nesanton** [comment]
@evgeni, what do you think of ^^^

**Comment by nesanton** [comment]
This will totally screw all unmerged PRs ;)

**Comment by evgeni** [comment]
So bear with me while I read the diff, this PR does essentially the following:

* move spec initialization/parsing from ForemanEntityAnsibleModule to ForemanAnsibleModule
* `s/entity_spec/foreman_spec/g`
* `s/entity_dict/module_params/g`

So now non-Entity modules could use `foreman_spec`, but they so far don't yet, correct? Not saying it has to be in this PR, just making sure I did not overlook anything.

That all said, the changes look sane to me.

@bagasse and @mdellweg your opinions as heavy cerators/users of the spec are appreciated :)

**Comment by nesanton** [comment]
> * move spec initialization/parsing from ForemanEntityAnsibleModule to ForemanAnsibleModule
> * `s/entity_spec/foreman_spec/g`
> * `s/entity_dict/module_params/g`

100% precision.

I've tried merging it with templates_sync and replacing arg_spec to forman_spec. Seems to work

**Comment by nesanton** [comment]
Here you go

**Comment by bagasse** [comment]
It's ok for me too :)

**Comment by mdellweg** [comment]
Maybe a little late to the party, but i like it!

**Pull request merged**